docs/using.md

```diff
@@ -183,22 +242,45 @@ This dictionary can also be deserialized:
 Token.from_dict(tokens[1].as_dict())
 ```
 
-In some use cases `nest_tokens` may be useful, to collapse the opening/closing tokens into single tokens:
+### Creating a syntax tree
+
+```{versionchanged} 0.7.0
+`nest_tokens` and `NestedTokens` are deprecated and replaced by `SyntaxTreeNode`.
+```
+
+In some use cases it may be useful to convert the token stream into a syntax tree,
+with opening/closing tokens collapsed into a single token that contains children.
 
 ```{code-cell}
-from markdown_it.token import nest_tokens
-nested_tokens = nest_tokens(tokens)
-[t.type for t in nested_tokens]
+from markdown_it.tree import SyntaxTreeNode
+
+md = MarkdownIt("commonmark")
+tokens = md.parse("""
+# Header
+
+Here's some text and an image ![title](image.png)
+
+1. a **list**
+
+> a *quote*
+""")
+
+node = SyntaxTreeNode.from_tokens(tokens)
```

Comment:
Just a heads-up, I made a PR (https://github.com/executablebooks/markdown-it-py/pull/132) which would change this syntax to `node = SyntaxTreeNode(tokens)` so if we merge that then one of the PRs needs to adapt depending on merge order.

Reply:
I'll have a look